### DISCUSSION TITLE
Subresource Integrity (`integrity` attribute)

### DISCUSSION THREAD

**Discussion by LeaVerou** [Devographics/surveys#145] [State of HTML Features]
A security feature to verify that fetched resources have not been tampered with.

### Code sample

```html
<script src="https://code.jquery.com/jquery-3.js" crossorigin
 integrity="sha384-q8i/X+965DzO0Kp4H+8aPi6jizo"></script>
```

### MDN URL:

https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity

### Specification URL:

https://www.w3.org/TR/SRI/#the-integrity-attribute

### Any other links?

- https://css-tricks.com/securing-your-website-with-subresource-integrity/
- https://www.smashingmagazine.com/2019/04/understanding-subresource-integrity/

### How stable is this feature?

Stable (Widely supported for several years)

### Type of feature

HTML attribute

### What focus area does this feature fall into?

Security & Privacy

**Comment by lukewarlow** [comment]
https://github.com/Devographics/surveys/discussions/108 seems to duplicate this?

**Reply by LeaVerou** [reply]
Whoops, good point, thanks! Deleted the other one.
